portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -103,9 +111,12 @@ private void printHTMLWithAllMemes(HttpServletResponse response, PreparedQuery r
       String argURL = (String) entity.getProperty("url");
       String argComment = (String) entity.getProperty("comment");
       long timestamp = (long) entity.getProperty("timestamp");
+
+      String userEmail = (String) entity.getProperty("userEmail");
       
       response.getWriter().println("\n <p> ### ### ### </p>\n");
       response.getWriter().println("\n <p> ID: " + id + "</p>");
+      response.getWriter().println("\n <p> User: " + userEmail + "</p>");
```

Comment:
Generating HTML with String constructions are hard to read and maintain.
My suggestion is to adopt template engine like Free Maker: https://freemarker.apache.org/